lib/trino-parquet/src/main/java/io/trino/parquet/writer/ParquetWriters.java

```diff
@@ -126,20 +127,20 @@ static PrimitiveValueWriter getValueWriter(ValuesWriter valuesWriter, Type type,
         }
         if (type instanceof TimestampType) {
             if (parquetType.getPrimitiveTypeName().equals(PrimitiveType.PrimitiveTypeName.INT96)) {
-                checkArgument(parquetTimeZone.isPresent(), "parquetTimeZone must be provided for INT96 timestamps");
+                checkArgument(parquetTimeZone.isPresent(), "parquetTimeZone must be provided for timestamps");
                 return new Int96TimestampValueWriter(valuesWriter, type, parquetType, parquetTimeZone.get());
             }
             if (TIMESTAMP_MILLIS.equals(type)) {
                 verifyParquetType(type, parquetType, TimestampLogicalTypeAnnotation.class, isTimestamp(LogicalTypeAnnotation.TimeUnit.MILLIS));
-                return new TimestampMillisValueWriter(valuesWriter, type, parquetType);
+                return new TimestampMillisValueWriter(valuesWriter, type, parquetType, parquetTimeZone);
```

Comment:
I've had a re-think about the writer changes. Now that we're deciding on timezone conversion based on parquet annotation rather than just the time zone config, we don't have the problem of data not round tripping properly in tests.
I had also found that in Hive 4, when int64 timestamps (`set hive.parquet.write.int64.timestamp=true;`) are written, they're not adjusted to UTC.
Let's drop the writer changes for now.

Reply:
I think without https://github.com/trinodb/trino/pull/22577/files#diff-3b64954c48d2c0059a9156e9ef84a6b66d58c436ea593753a78cd27a604c92feR128-R328 changes, writer will write time without converting to to UTC and reader will consider that time in UTC and then, convert to local, do we need to make any other changes?

Reply:
Let's say we start by reading adjustedToUTC INT64 timestamp, we will convert from UTC to local. If we write it out as iceberg or delta, then it's INT64 timestamps in local time, which is expected for iceberg/delta tables. It we write it out as hive table, then it will be int96 timestamp which is always converted to UTC.
So I don't think there is a problem with skipping the writer changes.

Reply:
Thanks for clarifying the round trip for table formats. However, Parquet round trip test `AbstractTestParquetReader#testTimestamp` is failing without the mentioned change.

Reply:
i pushed a fixup commit which removes the writer changes, AbstractTestParquetReader#testTimestamp is passing

Reply:
I didn't revert the `isAdjustedToUTC` changes in `ParquetSchemaConverter`